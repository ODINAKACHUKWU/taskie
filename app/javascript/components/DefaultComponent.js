import React from "react";

function DefaultComponent() {
  return <h1 className="welcome-text">Welcome to the Taskie app!</h1>;
}

export default DefaultComponent;
