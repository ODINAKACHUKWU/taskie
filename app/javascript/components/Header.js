import React from "react";
import PropTypes from "prop-types";

const defaultProps = {
  title: "Taskie",
};

function Header({ title = defaultProps.title, action }) {
  return (
    <div className="header-container">
      <div className="title">{title}</div>
      <div className="action-button">{action}</div>
    </div>
  );
}

Header.propTypes = {
  title: PropTypes.string.isRequired,
  action: PropTypes.element,
};

export default Header;
