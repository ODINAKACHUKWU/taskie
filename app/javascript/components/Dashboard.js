import React from "react";
import PropTypes from "prop-types";
import Header from "./Header";
import DefaultComponent from "./DefaultComponent";

const defaultProps = {
  children: <DefaultComponent />,
};

function Dashboard({ title, action, children = defaultProps.children }) {
  return (
    <div className="dashboard-container">
      <Header title={title} action={action} />
      <div className="main">{children}</div>
    </div>
  );
}

Dashboard.propTypes = {
  title: PropTypes.string.isRequired,
  action: PropTypes.element,
  children: PropTypes.element.isRequired,
};

export default Dashboard;
