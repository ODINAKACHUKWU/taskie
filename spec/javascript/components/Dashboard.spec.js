import "jsdom-global/register";
import React from "react";
import { shallow, mount } from "enzyme";
import toJSON from "enzyme-to-json";
import Dashboard from "../../../app/javascript/components/Dashboard";
import DefaultComponent from "../../../app/javascript/components/DefaultComponent";
import Header from "../../../app/javascript/components/Header";

const defaultProps = {
  title: "Taskie",
  action: null,
  children: <DefaultComponent />,
};

describe("rendering Dashboard component", () => {
  it("renders Dashboard component without crashing", () => {
    shallow(
      <Dashboard title={defaultProps.title}>{defaultProps.children}</Dashboard>
    );
  });

  it("renders Header component", () => {
    const wrapper = mount(
      <Dashboard title={defaultProps.title}>{defaultProps.children}</Dashboard>
    );
    const header = <Header title={defaultProps.title} />;
    expect(wrapper.contains(header)).toEqual(true);
  });
});

describe("passing props", () => {
  it("accepts title, action and children props", () => {
    const wrapper = mount(
      <Dashboard title={defaultProps.title} action={defaultProps.action}>
        {defaultProps.children}
      </Dashboard>
    );

    expect(wrapper.props().title).toEqual(defaultProps.title);
    expect(wrapper.props().action).toEqual(null);
    expect(wrapper.props().children).toEqual(defaultProps.children);
  });
});

describe("snapshot", () => {
  it("Dashboard snapshot", () => {
    const tree = shallow(
      <Dashboard title={defaultProps.title}>{defaultProps.children}</Dashboard>
    );

    expect(toJSON(tree)).toMatchSnapshot();
  });
});
