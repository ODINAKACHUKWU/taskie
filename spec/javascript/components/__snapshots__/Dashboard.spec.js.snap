// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshot Dashboard snapshot 1`] = `
<div
  className="dashboard-container"
>
  <Header
    title="Taskie"
  />
  <div
    className="main"
  >
    <DefaultComponent />
  </div>
</div>
`;
